cpp/es2.1/dateEdoardo.cpp

```diff
@@ -0,0 +1,153 @@
+#include<iostream>
+
+using namespace std;
+
+class Date {
+    private:
+        int day, month, year;
+
+    public:
+        Date() {day=1; month=1; year=1970;}
+        Date(int day, int month, int year) {this->day=day; this->month=month; this->year=year;}
+        int getDay() {return day;}
+        int getMonth() {return month;}
+        int getYear() {return year;}
+        friend ostream& operator<<(ostream &out, Date &pt);
+        friend bool operator==(Date &first, Date &second);
+        friend void addDays(int n, Date d);
+        friend void addMonth(int n, Date d);
+        friend void addYear(int n, Date d);
+        friend void adjust(Date d);
```

Comment:
Tutte queste sono funzioni libere che accedono all'implementazione privata, perché stai usando `friend`. Vanno trasformati in metodi di classe.